### ISSUE TITLE
[EVENT] Callysto

### ISSUE THREAD

**Issue by damianavila** [2i2c-org/infrastructure#1918]
### Summary

We have a series of workshops where various groups of users will be logging and and trying out the service.
Ref: https://github.com/2i2c-org/infrastructure/issues/1678#issuecomment-1302631346

- **Community Representative:** @ianabc @byrcyb
- **Event begin:** (dates already updated)
  - Nov 24th ~ 50-60 users
  - Nov 25th ~ 180-190 users
  - December 5th ~200 students
  - December 6th ~ 120 users
  - December 9th ~ 30 users (new)
  - December 16th (to be confirmed) ~30 students
- **Event end:**
- **Active times:** Starting at 9 am (timezone?)
- **Number of attendees:** See above on the dates section
- [**Hub Events Calendar**](https://calendar.google.com/calendar/u/2?cid=Y19rdDg0c2g3YW5tMHNsb2NqczJzdTNqdnNvY0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t) (already added after dates confirmation)

### Hub info

- **Hub URL**: 2i2c.callysto.ca
- **Hub decommisioned after event?**: NO

### Task List

**Before the event**

- [x] Dates confirmed with the community representative and added to Hub Events Calendar.
- [x] Quotas from the cloud provider are high-enough to handle expected usage.
- [x] **One week before event** Hub is running.
- [x] Confirm with Community Representative that their workflows function as expected.
  - <details>
    <summary>👉Template message to send to community representative</summary>
    
    ```
    Hey {{ COMMUNITY REPRESENTATIVE }}, the date of your event is getting close!
    
    Could you please confirm that your hub environment is ready-to-go, and matches your hub's infrastructure setup, by ensuring the following things:
    - [ ] Confirm that the "Event Info" above is correct
    - [ ] On your hub: log-in and authentication works as-expected
    - [ ] `nbgitpuller` links you intend to use resolve properly
    - [ ] Your notebooks and content run as-expected
    ```
    
    </details>  
- [ ] **1 day before event**, either a separate nodegroup is provisioned for the event or the cluster is scaled up.

**During and after event**

- [x] Confirm event is finished.
- [ ] Nodegroup created for the hub is decommissioned / cluster is scaled down.
- [ ] Hub decommissioned (if needed).
- [x] Debrief with community representative.
  - <details>
    <summary>👉Template debrief to send to community representative</summary>
      
     ```
     Hey {{ COMMUNITY REPRESENTATIVE }}, your event appears to be over 🎉
     
     We hope that your hub worked out well for you! We are trying to understand where we can improve our hub infrastructure and setup around events, and would love any feedback that you're willing to give. Would you mind answering the following questions? If not, just let us know and that is no problem!
  
     - Did the infrastructure behave as expected?
     - Anything that was confusing or could be improved?
     - Any extra functionality you wish you would have had?
     - Could you share a story about how you used the hub?

     - Any other feedback that you'd like to share?
     
     ```

    </details>

**Comment by damianavila** [comment]
There is currently a long discussion happening on an associated issue: https://github.com/2i2c-org/infrastructure/issues/1678.

* Self-warming of the cluster will be allowed as per this comment: https://github.com/2i2c-org/infrastructure/issues/1678#issuecomment-1304303297. We need to confirm they can actually do it.

* Authentication is being discussed, once we get into an agreement, that decision should be ported over here.

@ianabc @byrcyb, can you please re-confirm the dates? Thanks!

**Comment by byrcyb** [comment]
Yes, here are the confirmed dates for upcoming hackathons where we intend to use the 2i2c callysto hub:
* Nov 24th ~ 50-60 users
* Nov 25th ~ 180-190 users
* December 6th ~ 120 users
* December 9th ~ 30 users (new)

The November 18th hackathon is happening but we are going to use our everyday hub (hub.callysto.ca) for this event.

**Comment by ianabc** [comment]
I think we have a plan for authentication (see #1678) for at least the next two workshops. I was also able to get in to the google cloud console and warm up the nodes as discussed. I want to make sure I'm doing that the right way though, so @damianavila could you confirm the steps below

1. Set a new value for the autoscaler minimum number of nodes (1)
2. Set the actual number of nodes (also 1).
3. Check node is added and in ready state
4. Log in, use it, then shut down my pod
5. Set autoscaler minimum back to 0
6. Confirm node is removed

The only issue I had was that I had to wait while our image was pulled (it's a bit big so ~3mins) is that normal? Or should the ImagePuller have done that for me on node creation?

**Comment by yuvipanda** [comment]
@ianabc I opened https://github.com/2i2c-org/infrastructure/pull/1943 to enable the prePuller, should allow pre-warming to also pre-pull images.

**Comment by damianavila** [comment]
> I want to make sure I'm doing that the right way though, so @damianavila could you confirm the steps below

The steps seem correct to me. 
@yuvipanda do you have any other recommendations for them?
Btw, Yuvi already addressed the prepuller piece in referenced PRs.

**Comment by byrcyb** [comment]
As an update, we've added one and possibly a second hackathon event to the schedule:
- December 5th ~200 students
- December 16th (to be confirmed) ~30 students

**Comment by yuvipanda** [comment]
@byrcyb how did the last one go?

**Comment by byrcyb** [comment]
> @byrcyb how did the last one go?

We had some surprises 😅 - there were no issues with the 2i2c infrastructure and authentication worked smoothly for those who could resolve the url. As well, the warm-up worked as we had hoped with no node start-up time experienced by the participants. We had troubles, as we think the school district unexpectedly introduced a firewall rule, after we had completed testing prior to the event, that prevented students from resolving 2i2c.callysto.ca. So, we had to scramble and provide some nbgitpuller links that pointed to our old hub.callysto.ca. After an hour or so, we were able to work with the school's network team to get things fixed so the students could use the 2i2c hub ... so it worked out in the end!
